### PR TITLE
changes based on feedback & bugs I've noticed

### PR DESCRIPTION
**Pull request by Nathan-Binkley**

### CODE REVIEW

File: bot/cogs/grades_data/grades_cog.py [hunk 1]
```diff
@@ -33,19 +33,48 @@ class gradesCog(commands.Cog):
...
         self.fileLines = []
+
         self.master_list = {}
         self.master_prof_list = {}
 
-    def process_Search(self, orig_query: str) -> str: #Primary search function and processing for a query. A query is generally defined by 
+        self.load_files()
+
+    def load_files(self):
+        temp = self.initialize('2014') # Store 2014
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:43]: this whole thing can be a loop that takes an iterable of strings, probably an iterable that is passed into the function

File: bot/cogs/grades_data/grades_cog.py [hunk 2]
```diff
@@ -168,9 +198,9 @@ def getFirstLast(self, Name):
         Last = fml[0]
         return First + ' ' + Last
 
-    def process_profQuery(self, Name):
+    def process_profQuery(self, Name, year=2014):
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:201]: lets pull this magic number out into a module constant called MIN_YEAR and use that, just easier to keep track of that way and avoids magic numbers

File: bot/cogs/grades_data/grades_cog.py [hunk 3]
```diff
@@ -249,12 +277,12 @@ async def grades(self, ctx, course, year=2014):
...
             embed = discord.Embed(title="Grades", color=Colors.Error)
-            result = 'That\'s not a course\n Are you sure you used the proper notation (ex: cpsc-2120)'
+            result = 'That\'s not a course\n Are you sure you used the proper notation (ex: cpsc-2120)?'
             embed.add_field(name="ERROR: Course doesn't exist", value=result, inline=False)
         
-        except FileNotFoundError:
+        except FileNotFoundError as e:
             embed = discord.Embed(title="Grades", color=Colors.Error)
-            result = "Either master.json or master_prof.json are not found in the proper directory"
+            result = f'Files `{e}` are not in directory. Reminder that the files go from 2014-2019!'
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:285]: Thought: if we make the previously mentioned iterable of strings for the load function an instance member we could dynamically create this string based off that, example 

`'f'Files {e} are not in directory. Reminder that the files go from {self.years[0]}-{self.years[-1]}!'`

File: bot/cogs/grades_data/grades_cog.py [hunk 4]
```diff
@@ -30,39 +30,23 @@ class gradesCog(commands.Cog):
...
 
+    MIN_YEAR = 2014
+
     def __init__(self, bot):
         self.bot = bot
         self.fileLines = []
 
         self.master_list = {}
         self.master_prof_list = {}
+        self.MIN_YEAR = 2014
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:41]: you dont need this, this is a separate instance const, we can use just the module const one

File: bot/cogs/grades_data/grades_cog.py [hunk 5]
```diff
@@ -266,23 +250,28 @@ async def grades(self, ctx, course, year=2014):
...
+                result = self.go(course, year)
+                embed.add_field(name="Result", value=result, inline=False)
+                prefix = await self.bot.get_prefix(ctx)
+                if isinstance(prefix, list):
+                    prefix = prefix[0]
+                exp = f'Type `{prefix}help grades` for more information' # NOQA
+                embed.add_field(name='Explanation', value=exp)
+            else:
+                embed = discord.Embed(title="Grades", color=Colors.Error)
+                result = f'Year not available. Reminder that years go from {self.year_list[0]}-{self.year_list[len(self.year_list)-1]}'
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:264]: `{self.year_list[0]}-{self.year_list[len(self.year_list)-1]}` can just be 

`{self.year_list[0]}-{self.year_list[-1]}`

File: bot/cogs/grades_data/grades_cog.py [hunk 6]
```diff
@@ -266,23 +250,28 @@ async def grades(self, ctx, course, year=2014):
...
         except NotADirectoryError: # output if course doesn't exist
             embed = discord.Embed(title="Grades", color=Colors.Error)
             result = 'That\'s not a course\n Are you sure you used the proper notation (ex: cpsc-2120)?'
             embed.add_field(name="ERROR: Course doesn't exist", value=result, inline=False)
         
-        except FileNotFoundError as e:
+        except FileNotFoundError as e: #SHOULD NEVER THROW
             embed = discord.Embed(title="Grades", color=Colors.Error)
-            result = f'Files `{e}` are not in directory. Reminder that the files go from 2014-2019!'
+            result = f'Files `{e}` are not in directory. Reminder that the files go from {self.year_list[0]}-{self.year_list[len(self.year_list)-1]}!'
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:274]: same thing as above

**Comment by Nathan-Binkley** [comment]
Fixed lol

**Pull request merged**